src/bioutils/sequences.py

```diff
@@ -441,7 +453,12 @@ def translate_cds(seq, full_codons=True, ter_symbol="*"):
     protein_seq = list()
     for i in range(0, len(seq) - len(seq) % 3, 3):
         try:
-            aa = dna_to_aa1_lut[seq[i:i + 3]]
+            the_seq = seq[i:i + 3]
+            wildcard_nucleotides = ["B", "D", "H", "V", "N", "U", "W", "S", "M", "K", "R", "Y", "Z"]
+            if any([wildcard_base in the_seq for wildcard_base in wildcard_nucleotides]):
```

Comment:
As written, any wildcard cause the AA to be X. I think we can do better than that. For example, in a standard translation table, CUN ⇒ Leu, GCN ⇒ Ala,  GGN ⇒ Gly, AAY ⇒ Asn, etc. See overall comments for discussion.